### PR TITLE
Farewell to the Static API

### LINKED ISSUE #3952: Remove Static API code
Since we are closing Static API issues with "no development is being done on static API" (ex. https://github.com/dotnet/machinelearning/issues/1153#issuecomment-507014868), we should just remove the Static API code from the repo. If we need it again in the future, we can get it out of source control history. But there is no reason to build and maintain this code anymore.

### PR DESCRIPTION
**Pull request by artidoro**
Fixes #3952.

In this PR I 1. remove the static API code and 2. migrate the tests to the C# API (dynamic API) and 3. stop producing the static pipe nugets.

The commits are organized logically, which should simplify the review process.

### CODE REVIEW

**Comment by codemzs** [comment]
Were you thinking of removing references to dynamic API? and just renaming it to API?

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TreeFeaturization/FastForestBinaryFeaturizationWithOptions.cs [hunk 1]
```diff
@@ -58,7 +58,8 @@ public static void Example()
             };
 
             // Define the tree-based featurizer's configuration.
-            var options = new FastForestBinaryFeaturizationEstimator.Options
+            var options = new FastForestBinaryFeaturizationEstimator
+    .Options
```
Comment (codemzs) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TreeFeaturization/FastForestBinaryFeaturizationWithOptions.cs]: >.Options [](start = 4, length = 8)

was this crossing 85 character limit?

**Comment by codemzs** [comment]
> // The .NET Foundation licenses this file to you under the MIT license.

why is this file showing as modified? I don't see any changes.

---
Refers to: test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs:2 in ee68b9c. [](commit_id = ee68b9caa16cb25c9560b9259081e0b7797176bb, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TreeFeaturization/FastForestBinaryFeaturizationWithOptions.cs (hunk 1 above)
Comment (artidoro) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TreeFeaturization/FastForestBinaryFeaturizationWithOptions.cs]: Sorry that's a mistake, I'll revert that. Thanks for pointing it out.

---
In reply to: [304142959](https://github.com/dotnet/machinelearning/pull/4009#discussion_r304142959) [](ancestors = 304142959)

**Comment by artidoro** [comment]
> // The .NET Foundation licenses this file to you under the MIT license.

Good point, I'll remove it from the PR.

---
In reply to: [512007726](https://github.com/dotnet/machinelearning/pull/4009#issuecomment-512007726) [](ancestors = 512007726)

---
Refers to: test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs:2 in ee68b9c. [](commit_id = ee68b9caa16cb25c9560b9259081e0b7797176bb, deletion_comment = False)

**Comment by artidoro** [comment]
I can do it in this PR or in another PR. It might be a more documentation focused PR, so I was not thinking to have in this PR initially. What do you think?

---
In reply to: [512006889](https://github.com/dotnet/machinelearning/pull/4009#issuecomment-512006889) [](ancestors = 512006889)

File: src/Microsoft.ML.Analyzer/TypeIsSchemaShapeAnalyzer.cs [hunk 2]
```diff
@@ -2,420 +2,19 @@
...
-using System.Linq;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace Microsoft.ML.Analyzer
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class TypeIsSchemaShapeAnalyzer : DiagnosticAnalyzer
```
Comment (eerhardt) [inline src/Microsoft.ML.Analyzer/TypeIsSchemaShapeAnalyzer.cs]: What's the value of this empty Analyzer? Shouldn't it just be deleted?

File: test/Microsoft.ML.OnnxTransformerTest/DnnImageFeaturizerTest.cs [hunk 3]
```diff
@@ -89,28 +85,25 @@ void TestDnnImageFeaturizer()
...
-                .Append(row => (
-                    row.name,
-                    data_0: row.imagePath.LoadAsImage(imageFolder).Resize(imageHeight, imageWidth).ExtractPixels(interleave: true)))
-                .Append(row => (row.name, output_1: row.data_0.DnnImageFeaturizer(m => m.ModelSelector.ResNet18(m.Environment, m.OutputColumn, m.InputColumn))));
+            var pipe = ML.Transforms.LoadImages("data_0", imageFolder, "imagePath")
+                .Append(ML.Transforms.ResizeImages("data_0", imageHeight, imageWidth))
+                .Append(ML.Transforms.ExtractPixels("data_0", interleavePixelColors: true))
+                .Append(ML.Transforms.DnnFeaturizeImage("output_1", m => m.ModelSelector.ResNet18(m.Environment, m.OutputColumn, m.InputColumn), "data_0"));
 
-            TestEstimatorCore(pipe.AsDynamic, data.AsDynamic);
```
Comment (eerhardt) [inline test/Microsoft.ML.OnnxTransformerTest/DnnImageFeaturizerTest.cs]: Do we still want to call `TestEstimatorCore` in this test?

**Comment by eerhardt** [comment]
> namespace FakeStaticPipes

Shouldn't this file be deleted?

---
Refers to: test/Microsoft.ML.StaticPipelineTesting/StaticPipeFakes.cs:16 in ee68b9c. [](commit_id = ee68b9caa16cb25c9560b9259081e0b7797176bb, deletion_comment = False)

File: test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs [hunk 4]
```diff
@@ -1,121 +0,0 @@
...
-            var path = GetDataPath(TestDatasets.breastCancer.trainFilename);
-            var env = new MLContext();
-            var data = TextLoaderStatic.CreateLoader(env, ctx => (
-                floatScalar: ctx.LoadFloat(1),
-                floatVector: ctx.LoadFloat(2, 6),
-                stringScalar: ctx.LoadText(4),
-                stringVector: ctx.LoadText(5, 7)
-            )).Load(path);
-
-            var enum1 = data.AsDynamic.GetColumn<float>(data.AsDynamic.Schema["floatScalar"]).ToArray();
```
Comment (eerhardt) [inline test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs]: Do we still want these "dynamic" GetColumn tests?

File: src/Microsoft.ML.Analyzer/TypeIsSchemaShapeAnalyzer.cs (hunk 2 above)
Comment (artidoro) [reply src/Microsoft.ML.Analyzer/TypeIsSchemaShapeAnalyzer.cs]: You are right, I though that this would give problems with the API compat since it is a public class, but I removed it altogether as you were suggesting and it does not complain. I guess it's because it's an analyzer!

---
In reply to: [304492042](https://github.com/dotnet/machinelearning/pull/4009#discussion_r304492042) [](ancestors = 304492042)

**Comment by artidoro** [comment]
> namespace FakeStaticPipes

Yes you are right, I had removed it from the solution but not the files from the repo.

---
In reply to: [512346284](https://github.com/dotnet/machinelearning/pull/4009#issuecomment-512346284) [](ancestors = 512346284)

---
Refers to: test/Microsoft.ML.StaticPipelineTesting/StaticPipeFakes.cs:16 in ee68b9c. [](commit_id = ee68b9caa16cb25c9560b9259081e0b7797176bb, deletion_comment = False)

File: test/Microsoft.ML.OnnxTransformerTest/DnnImageFeaturizerTest.cs (hunk 3 above)
Comment (artidoro) [reply test/Microsoft.ML.OnnxTransformerTest/DnnImageFeaturizerTest.cs]: Yes thanks for pointing that out, I'll add it back.

---
In reply to: [304496249](https://github.com/dotnet/machinelearning/pull/4009#discussion_r304496249) [](ancestors = 304496249)

File: test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs (hunk 4 above)
Comment (artidoro) [reply test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs]: Yes we should keep these, you are right, I'll just remove the part that's static api only.

---
In reply to: [304498224](https://github.com/dotnet/machinelearning/pull/4009#discussion_r304498224) [](ancestors = 304498224)

File: docs/samples/Microsoft.ML.Samples/Microsoft.ML.Samples.csproj [hunk 5]
```diff
@@ -73,11 +72,6 @@
     </None>
     <PackageReference Include="Microsoft.ML.TensorFlow.Redist" Version="0.10.0" />
     
-    <ProjectReference Include="..\..\..\src\Microsoft.ML.Analyzer\Microsoft.ML.Analyzer.csproj">
```
Comment (eerhardt) [inline docs/samples/Microsoft.ML.Samples/Microsoft.ML.Samples.csproj]: Why this change?

**Comment by artidoro** [comment]
Thank you for reviewing!
